include/mbgl/style/layer.hpp

```diff
@@ -117,14 +111,12 @@ class Layer : public mbgl::util::noncopyable {
     void setMaxZoom(float) const;
 
     // Private implementation
-    const std::unique_ptr<Impl> baseImpl;
+    const std::shared_ptr<Impl> baseImpl;
```

Comment:
Per chat: good solution is probably to continue to pass copies of `Layer`s to the worker, rather than `RenderLayer`s. The worker can then create its own `RenderLayer`s, allowing us to preserve the invariant "`RenderLayers` never outlive their associated `Layer`" and avoid (unintentionally) sharing a `Layer::Impl` between the main thread and worker.